src/Components/Input.re

```diff
@@ -75,27 +76,40 @@ module Constants = {
   let selectionOpacity = 0.75;
 };
 
-module Styles = {
-  let defaultPlaceholderColor = Colors.grey;
-  let defaultCursorColor = Colors.black;
-  let defaultSelectionColor = Color.hex("#42557b");
+module Colors = {
+  let color = key => {
+    let key = ColorTheme.key(key);
+
+    theme =>
+      ColorTheme.Colors.get(key, theme)
+      |> Option.value(~default=Colors.transparentBlack);
+  };
 
-  let default =
+  let foreground = color("input.foreground");
+  let background = color("input.background");
+  let border = color("input.border");
+  let placeholderForeground = color("input.placeholderForeground");
+  let selection = color("selection.background");
+};
```

Comment:
Am example of a non-app-specific component using theme colors. This wasn't possible before because colors were hard-coded into an app-specific type, and so had to be passed in one-by one. This makes the input component much nicer to use. Of course there's still plenty of that around with Revery components. Perhaps we should move the `ColorTheme` infrastructure to Revery though.